src/pip/_internal/utils/deprecation.py

```diff
@@ -54,8 +55,14 @@ def install_warning_logger():
         warnings.showwarning = _showwarning
 
 
-def deprecated(reason, replacement, gone_in, issue=None):
-    # type: (str, Optional[str], Optional[str], Optional[int]) -> None
+def deprecated(
+    reason,
+    replacement,
+    gone_in,
+    gone_in_message=DEPRECATION_GONE_IN_MESSAGE,
```

Comment:
Is this argument realy needed? `DEPRECATION_MSG_PREFIX` is not configurable; who would need to configure this?

Reply:
It depends on what you mean by configuring though. If you just need to set this to an empty string, a global variable should be enough. The only usage I can think of needing this as an argument is you only want to remove `gone_in` from *some* deprecation messages but not all; when and why would you need to do that?

Reply:
> The only usage I can think of needing this as an argument is you only want to remove gone_in from some deprecation messages but not all; when and why would you need to do that?

Indeed, that's exactly what I want.

The reason is that when trying to use the normal template, I was forced to use very awkward phrasing. See https://github.com/pypa/pip/pull/10129#issuecomment-875961171 and https://github.com/pypa/pip/pull/10129#issuecomment-875980520.

Reply:
> The only usage I can think of needing this as an argument is you only want to remove gone_in from some deprecation messages but not all; when and why would you need to do that?

Indeed, that's exactly what I want.

The reason is that when trying to use the normal template, I was forced into very awkward phrasing. See https://github.com/pypa/pip/pull/10129#issuecomment-875961171 and https://github.com/pypa/pip/pull/10129#issuecomment-875980520. Thus for the occasional situation which requires a more elaborate explanation (namely in the case above), I would like to be able to disable it.